gpio: bcm-kona: remove unneeded platform_set_drvdata() call

CoardosAndrei opened the pull request:
The platform_set_drvdata() was needed when the driver had an explicit remove function.
That function got removed a while back, so we don't need to keep a pointer (on 'dev->driver_data') for the private data of the driver anymore.

CoardosAndrei committed:
gpio: bcm-kona: remove unneeded platform_set_drvdata() call

The platform_set_drvdata() was needed when the driver had an explicit
remove function.
That function got removed a while back, so we don't need to keep a pointer
(on 'dev->driver_data') for the private data of the driver anymore.

Signed-off-by: Andrei Coardos <traiandrei@yahoo.com>

CoardosAndrei committed:
gpio: eic-sprd:  remove unneeded platform_set_drvdata() call

remove unneeded platform_set_drvdata() call
The platform_set_drvdata() was needed when the driver had an explicit
remove function.
That function got removed a while back, so we don't need to keep a pointer
(on 'dev->driver_data') for the private data of the driver anymore.

Signed-off-by: Coardos Andrei <traiandrei@yahoo.com>

commodo commented on drivers/gpio/gpio-eic-sprd.c:
```diff
@@ -653,7 +653,6 @@ static int sprd_eic_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	platform_set_drvdata(pdev, sprd_eic);
```
your `Signed-off-by` tag is  with ` Coardos Andrei <traiandrei@yahoo.com>`
maybe change it to `Andrei Coardos  <traiandrei@yahoo.com>` ?

commodo replied:
same comment [as this one](https://github.com/commodo/linux/pull/14/commits/a3d1e02ae2d9d03a27c59fb2986822d62c57123a#r1263002893)

the driver was added in 2018;
and the `platform_set_drvdata()` call is just a copy+paste slip-up

commodo commented on drivers/gpio/gpio-bcm-kona.c:
```diff
@@ -597,7 +597,6 @@ static int bcm_kona_gpio_probe(struct platform_device *pdev)
 		return -ENOMEM;
 
 	kona_gpio->pdev = pdev;
-	platform_set_drvdata(pdev, kona_gpio);
```
bad news for this commit;
we'll need to change the description a bit;
`platform_set_drvdata()` was added from the beginning, but never used;

it looks like it was a copy+paste error; someone copied the driver, and just did not remove it;

commit description should be more llike 
```
The platform_set_drvdata() call was never used, ever since the driver was
originally added (back in 2013).
It looks like this was a copy+paste line of code which was taken from
another driver.
```

CoardosAndrei committed:
gpio: exar: remove unneeded platform_set_drvdata() call

The platform_set_drvdata() was needed when the driver had an explicit
remove function.
That function got removed a while back, so we don't need to keep a pointer
(on 'dev->driver_data') for the private data of the driver anymore.

Reviewed-by: Alexandru Ardelean <alex@shruggie.ro>
Signed-off-by: Andrei Coardos <traiandrei@yahoo.com>

CoardosAndrei committed:
gpio: logicvc: remove unneeded platform_set_drvdata() call.

In the gpio-logicvc driver found in drivers/gpio, the call to platform_set_drvdata was removed. This function call was found to be unnecessary as the associated structure, 'logicvc_gpio', is defined within the same file and doesn't utilize a getter function. Therefore, it only sets values and does not retrieve them. This change simplifies the code without altering functionality.

Reviewed-by: Alexandru Ardelean <alex@shruggie.ro>
Signed-off-by: Andrei Coardos <traiandrei@yahoo.com>
